apis/apiextensions/v1/composition_revision_types.go

```diff
@@ -38,7 +38,9 @@ const (
 type CompositionRevisionSpec struct {
 	// CompositeTypeRef specifies the type of composite resource that this
 	// composition is compatible with.
-	// +immutable
+	// +kubebuilder:validation:Optional
```

Comment:
Why mark this field optional? It isn't.

I don't think any of the fields you've added the `Optional` comment marker to are optional.

Reply:
@negz I have fixed those fields that are not supposed to be optional